dft/src/butterflies.rs

```diff
@@ -12,15 +15,46 @@ pub trait Butterfly<F: Field>: Copy + Send + Sync {
     fn apply_to_rows(&self, row_1: &mut [F], row_2: &mut [F]) {
         let (shorts_1, suffix_1) = F::Packing::pack_slice_with_suffix_mut(row_1);
         let (shorts_2, suffix_2) = F::Packing::pack_slice_with_suffix_mut(row_2);
-        assert_eq!(shorts_1.len(), shorts_2.len());
-        assert_eq!(suffix_1.len(), suffix_2.len());
+        debug_assert_eq!(shorts_1.len(), shorts_2.len());
+        debug_assert_eq!(suffix_1.len(), suffix_2.len());
         for (x_1, x_2) in shorts_1.iter_mut().zip(shorts_2) {
             self.apply_in_place(x_1, x_2);
         }
         for (x_1, x_2) in suffix_1.iter_mut().zip(suffix_2) {
             self.apply_in_place(x_1, x_2);
         }
     }
+
+    /// Like `apply_to_rows`, but out-of-place.
+    #[inline]
+    fn apply_to_rows_oop(
```

Comment:
return `(&mut [F], &mut [F])`?

Reply:
Actually since the caller is doing `chunks_mut` to get smaller subslices, I don't really see a nice way to change this that would improve the caller